File: 0048-rotate-image/0048-rotate-image.cpp
class Solution {
public:
    void transpose(vector<vector<int>>& m){
        int r=m.size();
        int c=m[0].size();
        
        for(int i=0;i<r;i++)
        {
            for(int j=i+1;j<c;j++)
            {
                swap(m[i][j],m[j][i]);
            }
        }
    }
    
    void reversed(vector<vector<int>>& m){
        for(int i=0;i<m.size();i++){
            reverse(m[i].begin(),m[i].end());
        }
    }
    void rotate(vector<vector<int>>& matrix) {
        transpose(matrix);
        reversed(matrix);
    }
};

File: 0048-rotate-image/0048-rotate-image.py
class Solution:
    

    def rotate(self, matrix: List[List[int]]) -> None:
        """
        Do not return anything, modify matrix in-place instead.
        """
        rows = len(matrix)
        matrix1 = list(map(list, zip(*matrix)))
        for i in range(rows):
            matrix[i] = matrix1[i][::-1]

    
        